arduino-ide-extension/src/browser/contributions/new-version-notification.ts

```diff
@@ -0,0 +1,29 @@
+import { injectable, inject } from 'inversify';
+import { MessageService } from '@theia/core/lib/common/message-service';
+import { FrontendApplicationContribution } from '@theia/core/lib/browser/frontend-application';
+import { UpdatesRetriever } from '../updates/updates-retriever';
+import { shell } from 'electron';
+
+const GO_TO_DOWNLOAD_PAGE = 'Go to download page...';
+/**
+ * Listens on `BoardsConfig.Config` changes, if a board is selected which does not
+ * have the corresponding core installed, it proposes the user to install the core.
+ */
+@injectable()
+export class NewVersionNotification implements FrontendApplicationContribution {
+    @inject(UpdatesRetriever)
+    private readonly updatesRetriever: UpdatesRetriever;
+
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+
+    async onStart(): Promise<void> {
+        if (await this.updatesRetriever.isUpdateAvailable()) {
```

Comment:
`await` will block the app startup for the fetch time. It would be better to do `this.updatesRetriever.isUpdateAvailable.then(available -> )`. What do you think?

Reply:
@fstasi @AlbyIanna 
please have a look at @kittaakos 's suggestion